### PR TITLE
feat: Add RTL support to modal header and related components

### PR DESCRIPTION
**Pull request by sakhawy**
## Description
Followup to [#525](https://github.com/django-cms/djangocms-admin-style/pull/525).

I touched related files for style consistency.

## Related resources
* [#525](https://github.com/django-cms/djangocms-admin-style/pull/525)
* [#521](https://github.com/django-cms/djangocms-admin-style/issues/521)

## Screenshots
Screenshot from 2024-04-01 10-45-16
Screenshot from 2024-04-01 10-45-28
Screenshot from 2024-04-01 10-45-33

## Checklist

* [X] I have opened this pull request against ``develop-4``
* [ ] I have added or modified the tests when changing logic
* [X] I have followed [the conventional commits guidelines](https://www.conventionalcommits.org/) to add meaningful information into the changelog
* [X] I have read the [contribution guidelines ](https://github.com/django-cms/django-cms/blob/develop/CONTRIBUTING.rst) and I have joined #workgroup-pr-review on [Slack](https://www.django-cms.org/slack) to find a “pr review buddy” who is going to review my pull request.

### CODE REVIEW

**Comment by joshyu** [comment]
@fsbraun  I found a post which is related to how to style RTL mode https://rtlstyling.com/posts/rtl-styling

File: cms/templates/cms/toolbar/toolbar.html [hunk 1]
```diff
@@ -4,7 +4,8 @@
      class="cms cms-reset
      {% if not user.is_authenticated %} cms-toolbar-auth{% endif %}
      {% if debug %} cms-toolbar-debug{% endif %}"
-     dir="ltr" data-touch-action="none">
+     lang="{{ LANGUAGE_CODE|default:"en-us" }}" dir="{{ LANGUAGE_BIDI|yesno:'rtl,ltr,auto' }}" 
+     data-touch-action="none">
```
Comment (fsbraun) [inline cms/templates/cms/toolbar/toolbar.html:8]:
```suggestion
     data-touch-action="none">
```
Do you think we can try to make the full toolbar RTL compatible? You've come so far... 

In that case, we do not need to set any language or direction, but just take it from the page...

File: cms/static/cms/sass/components/_structureboard.scss [hunk 2]
```diff
@@ -470,16 +470,15 @@
         right: 0;
```
Comment (fsbraun) [inline cms/static/cms/sass/components/_structureboard.scss:470]: I think the `right: ` in line 447 needs adjustment, too.

File: cms/static/cms/sass/components/_structureboard.scss [hunk 3]
```diff
@@ -234,11 +234,11 @@
             overflow: hidden;
```
Comment (fsbraun) [inline cms/static/cms/sass/components/_structureboard.scss:234]: In line 188, I believe this should be:  `padding-inline-start: $structure-draggable-inner-padding`

**Review by fsbraun — CHANGES_REQUESTED** [review]
@sakhawy This is cool stuff! Looks great.

I have two comments for the structure board.

Also, I can imagine going one last step further: Also make the toolbar RTL, i.e. the structure board on the left, the admin sidebar on the right and the menu from rtl. What do you think?

**Comment by sakhawy** [comment]
Thanks for the review @fsbraun! The toolbar is the very next thing on my list but is it good for the scope of this PR? Should I open up another one or just make the changes here?

File: cms/static/cms/sass/components/_structureboard.scss (hunk 2 above)
Comment (sakhawy) [reply cms/static/cms/sass/components/_structureboard.scss:470]: I'm not sure if this is the button that resizes the modal or not but if it is, I'll need so mirror the JavaScript code too when resizing.

Comment (fsbraun) [reply cms/static/cms/sass/components/_structureboard.scss:470]: No, that's the locked icon for text-enabled plugins in the structure board.

And yes, both the icon and the js code for resizing the modal would need to be adjusted, if you moved the resize button to the right.

**Review by fsbraun — APPROVED** [review]
LGTM!

**Comment by fsbraun** [comment]
@sakhawy Can you check if this change is still up-to-date?
https://github.com/django-cms/django-cms/blob/898a14a72100e2062cc2a389b2d49abe97bbba14/cms/templates/cms/toolbar/toolbar_with_structure.html#L37-L44

If so, I will merge and include the PR in django CMS 4.1.1.

**Comment by sakhawy** [comment]
@fsbraun I removed the `dir` attribute from all the child divs. The elements should now inherit the base direction from the html tag in `djangocms_frontend.html`.

**Pull request merged**